cmd/crypto/header.go

```diff
@@ -79,6 +96,27 @@ func (s3) ParseHTTP(h http.Header) (err error) {
 	return
 }
 
+// S3KMS represents AWS SSE-KMS. It provides functionality to
+// handle SSE-KMS requests.
+var S3KMS = s3KMS{}
+
+type s3KMS struct{}
+
+// IsRequested returns true if the HTTP headers indicates that
+// the S3 client requests SSE-KMS.
+func (s3KMS) IsRequested(h http.Header) bool {
+	if _, ok := h[SSEKmsID]; ok {
```

Comment:
In the case of `KMS` I thought that `X-Amz-Server-Side-Encryption-Aws-Kms-Key-Id` and `X-Amz-Server-Side-Encryption` are required and `X-Amz-Server-Side-Encryption`'s value will always be equal to `aws:kms`. `X-Amz-Server-Side-Encryption-Context` is the optional value, so we would not have to check for that. https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUT.html

Reply:
@kannappanr Correct. According to the documentation of this function it *indicates whether a S3 client requests SSE-KMS*. If a client specify `X-Amz-Server-Side-Encryption-Context` it indicates that SSE-KMS is requested - whether the (SSE-KMS) request is well-formed is not the responsibility of this function.

We have to detect a SSE request - even if it is malformed to respond with an appropriate error.